packages/aws-rfdk/lib/core/lambdas/nodejs/version-provider/handler.ts

```diff
@@ -0,0 +1,357 @@
+
+  readonly platform?: Platform;
+}
+
+export interface IUris {
+}
+
+export interface IDeadlineInstallerUris extends IUris {
+  readonly bundle: string;
+
+  readonly clientInstaller: string;
+
+  readonly repositoryInstaller: string;
+
+  readonly certificateInstaller?: string;
+}
+
+export interface IDeadlineDockerRecipeUris extends IUris {
+  readonly recipe: string;
+}
+
+export interface IVersionedUris {
+  /**
+   * The major version number.
+   */
+  readonly MajorVersion: string;
+
+  /**
+   * The minor version number.
+   */
+  readonly MinorVersion: string;
+
+  /**
+   * The release version number.
+   */
+  readonly ReleaseVersion: string;
+
+  /**
+   * The patch version number.
+   */
+  readonly PatchVersion: string;
+
+  /**
+   * The URLs to installers
+   */
+  readonly Uris: IUris;
+}
+
+export class VersionProvider extends SimpleCustomResource {
+  readonly indexFilePath: string|undefined;
+
+  constructor(indexFilePath?: string) {
+    super();
+    this.indexFilePath = indexFilePath;
+  }
+
+  /**
+   * @inheritdoc
+   */
+  /* istanbul ignore next */ // @ts-ignore
+  public validateInput(data: object): boolean {
+    return this.implementsIVersionProviderProperties(data);
+  }
+
+  /**
+   * @inheritdoc
+   */
+  // @ts-ignore  -- we do not use the physicalId
+  public async doCreate(physicalId: string, resourceProperties: IVersionProviderProperties): Promise<Map<Platform, IVersionedUris>> {
+    /* istanbul ignore next */
+    const indexJson = this.indexFilePath ? this.readInstallersIndex() : await this.downloadInstallerIndex();
+
+    const productSection = indexJson[resourceProperties.product];
+
+    if (!productSection) {
+      throw new Error(`Information about product ${resourceProperties.product} can't be found`);
+    }
+
+    let installers = new Map();
+    if (resourceProperties.platform) {
+      const versionedUris = await this.getUrisForPlatform(
+        resourceProperties.product,
+        productSection,
+        resourceProperties.platform,
+        resourceProperties.versionString);
+
+      if (versionedUris) {
+        installers.set(resourceProperties.platform, versionedUris);
+      }
+    } else {
+      Object.values(Platform).forEach(async p => {
+        const versionedUris = await this.getUrisForPlatform(
+          resourceProperties.product,
+          productSection,
+          p,
+          resourceProperties.versionString);
+
+        if (versionedUris) {
+          installers.set(p, versionedUris);
+        }
+      });
+    }
+
+    return installers;
+  }
+
+  /**
+   * @inheritdoc
+   */
+  /* istanbul ignore next */ // @ts-ignore
+  public async doDelete(physicalId: string, resourceProperties: IVersionProviderProperties): Promise<void> {
+    // Nothing to do -- we don't modify anything.
+    return;
+  }
+
+  private implementsIVersionProviderProperties(value: any): boolean {
+    if (!value || typeof(value) !== 'object') { return false; }
+
+    // TODO: Ignore case
```

Comment:
Can you either implement these TODO's or are you going to create issues to fix them.